crates/topos-tce-storage/src/lib.rs

```diff
@@ -32,62 +29,28 @@ pub(crate) mod rocks;
 mod tests;
 
 pub use client::StorageClient;
-pub use connection::Connection;
-pub use connection::ConnectionBuilder;
-
-#[cfg(feature = "rocksdb")]
-pub use rocks::RocksDBStorage;
 
 pub mod store;
 
 pub type PendingCertificateId = u64;
 
-/// Certificate index in the history of the source subnet
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Copy)]
-pub struct Position(pub u64);
-
-impl fmt::Display for Position {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.0)
-    }
-}
-
-impl Position {
-    const ZERO: Self = Self(0);
-
-    pub fn increment(self) -> Result<Self, PositionError> {
-        match self {
-            Self::ZERO => Ok(Self(1)),
-            Self(value) => value
-                .checked_add(1)
-                .ok_or(PositionError::MaximumPositionReached)
-                .map(Self),
-        }
-    }
-}
-
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct CertificateSourceStreamPosition {
+#[derive(Debug, Clone)]
+pub struct CertificateTargetStreamPosition {
+    pub target_subnet_id: SubnetId,
     pub source_subnet_id: SubnetId,
     pub position: Position,
 }
 
-impl From<SourceStreamPositionKey> for CertificateSourceStreamPosition {
-    fn from(value: SourceStreamPositionKey) -> Self {
-        CertificateSourceStreamPosition {
-            source_subnet_id: value.0,
-            position: value.1,
+impl From<TargetStreamPositionKey> for CertificateTargetStreamPosition {
```

Comment:
We could completelly remove `TargetStreamPositionKey`, looks the same as `CertificateTargetStreamPosition`